File: runtime/src/test/java/com/siimkinks/sqlitemagic/UtilsTest.java
package com.siimkinks.sqlitemagic;

import org.junit.Test;

import static com.google.common.truth.Truth.assertThat;
import static com.siimkinks.sqlitemagic.Utils.toByteArray;

public final class UtilsTest {
  @Test
  public void boxedByteArrayToUnboxed() {
    final byte[] expected = new byte[]{0x55, 0x66, 0x14};
    assertThat(toByteArray(new Byte[]{0x55, 0x66, 0x14})).isEqualTo(expected);
  }

  @Test
  public void unboxedByteArrayToBoxed() {
    final Byte[] expected = new Byte[]{0x55, 0x66, 0x14};
    assertThat(toByteArray(new byte[]{0x55, 0x66, 0x14})).isEqualTo(expected);
  }
}

File: runtime/src/test/kotlin/com/siimkinks/sqlitemagic/UtilsTest.kt
package com.siimkinks.sqlitemagic

import com.google.common.truth.Truth.assertThat
import com.siimkinks.sqlitemagic.Utils.toByteArray
import org.junit.Test

class UtilsTest {
  @Test
  fun `converts boxed byte array to unboxed`() {
    val expected = byteArrayOf(0x55, 0x66, 0x14)

    assertThat(toByteArray(arrayOf(0x55, 0x66, 0x14)))
      .isEqualTo(expected)
  }

  @Test
  fun `converts unboxed byte array to boxed`() {
    val expected = arrayOf<Byte>(0x55, 0x66, 0x14)

    assertThat(toByteArray(byteArrayOf(0x55, 0x66, 0x14)))
      .isEqualTo(expected)
  }
}
